operations/mimir-mixin/config.libsonnet

```diff
@@ -22,6 +22,43 @@
     // modify the job selectors in the dashboard queries.
     singleBinary: false,
 
+    // This is mapping between a Mimir component name and the regular expression that should be used
+    // to match its instance and container name. Mimir jsonnet and Helm guarantee that the instance name
+    // (e.g. Kubernetes Deployment) and container name always match, so it's safe to use a shared mapping.
+    //
+    // This mapping is intentionally local and can't be overridden. If the final user needs to customize
+    // dashboards and alerts, they should override the final matcher regexp (e.g. container_names or instance_names).
+    local componentNameRegexp = {
+      // Microservices deployment mode.
+      compactor: 'compactor',
+      alertmanager: 'alertmanager',
+      ingester: 'ingester',
+      distributor: 'distributor',
+      querier: 'querier',
+      query_frontend: 'query-frontend',
+      query_scheduler: 'query-scheduler',
+      ruler: 'ruler',
+      ruler_querier: 'ruler-querier',
+      ruler_query_frontend: 'ruler-query-frontend',
+      ruler_query_scheduler: 'ruler-query-scheduler',
+      store_gateway: 'store-gateway',
+      overrides_exporter: 'overrides-exporter',
+      gateway: '(gateway|cortex-gw|cortex-gw-internal)',
+
+      // Read-write deployment mode.
+      mimir_write: 'mimir-write',
+      mimir_read: 'mimir-read',
+      mimir_backend: 'mimir-backend',
+    },
+
+    // Some dashboards show panels grouping together multiple components of a given "path".
+    // This mapping configures which components belong to each group.
+    component_groups: {
+      write: ['distributor', 'ingester', 'mimir_write'],
+      read: ['query_frontend', 'querier', 'ruler_query_frontend', 'ruler_querier', 'mimir_read'],
+      backend: ['query_scheduler', 'ruler_query_scheduler', 'ruler', 'store_gateway', 'compactor', 'alertmanager', 'overrides_exporter', 'mimir_backend'],
+    },
```

Comment:
This isn't used in places other than this file. WDYT about making it a local variable like `componentNameRegexp` to reduce the scope?